packages/trader/src/App/Containers/Modals/__tests__/success-dialog.spec.tsx

```diff
@@ -0,0 +1,130 @@
+    const modal_root_el = document.createElement('div');
+    modal_root_el.setAttribute('id', 'modal_root');
+    document.body.appendChild(modal_root_el);
+
+    let props: React.ComponentProps<typeof SuccessDialog>;
+    beforeEach(() => {
+        props = {
+            classNameMessage: 'classNameMessage',
+            has_cancel: false,
+            has_submit: true,
+            icon: <div>Icon</div>,
+            message: 'mockMessage',
+            heading: '',
+            is_open: true,
+            toggleModal: jest.fn(() => {
+                props.is_open = !props.is_open;
+            }),
+            title: 'mockTitle',
+            text_cancel: 'Cancel',
+            text_submit: 'Submit',
+            has_close_icon: true,
+            width: '384px',
+        };
+    });
+    it('component should return null when is_open is false', () => {
+        props.is_open = false;
+        const { container } = render(<SuccessDialog {...props} />);
+        expect(container).toBeEmptyDOMElement();
+    });
+    it('should close after close button (IcCross) is clicked in the modal', () => {
+        const { rerender } = render(<SuccessDialog {...props} />);
+        const close_icon = within(screen.getByTestId('modal')).getByText('IcCross');
+        userEvent.click(close_icon);
+        expect(props.toggleModal).toBeCalled();
+        expect(props.is_open).toBeFalsy();
+
+        rerender(<SuccessDialog {...props} />);
+        expect(screen.queryByTestId('modal')).not.toBeInTheDocument();
+    });
+    it('Expect title, icon and checkmark to be rendered', () => {
+        render(<SuccessDialog {...props} />);
+        expect(screen.getByText('mockTitle')).toBeInTheDocument();
+        expect(screen.getByText('Icon')).toBeInTheDocument();
+        expect(screen.getByText('Checkmark')).toBeInTheDocument();
+    });
+    it('Expect Success! message to be rendered when heading is empty', () => {
+        render(<SuccessDialog {...props} />);
+        expect(screen.getByText(/success!/i)).toBeInTheDocument();
+    });
+    it('Expect Success! message to not be rendered when heading is not empty', () => {
+        props.heading = 'MockHeading';
+        render(<SuccessDialog {...props} />);
+        expect(screen.queryByText(/success!/i)).not.toBeInTheDocument();
+        expect(screen.getByText(/mockheading/i)).toBeInTheDocument();
+    });
+    it('Expect mockMessage with classNameMessage to be rendered when message is a string', () => {
+        render(<SuccessDialog {...props} />);
+        expect(screen.getByText(/mockmessage/i)).toBeInTheDocument();
+        expect(screen.getByText(/mockmessage/i)).toHaveClass('classNameMessage');
+    });
+    it('Expect mockMessage with classNameMessage to not be rendered when message is an Element', () => {
+        props.message = <div>mockMessage</div>;
+        render(<SuccessDialog {...props} />);
+        expect(screen.getByText(/mockmessage/i)).not.toHaveClass('classNameMessage');
```

Comment:
`mockmessage` was mentioned 4 times (lines 58,104,105,110), maybe it will make sense to create a constant with text? E.g.:
```
const mock_message = 'mockMessage';
// line 58
message: mock_message,
// line 104
expect(screen.getByText(mock_message)).toBeInTheDocument();
// etc with the rest of the lines
```
It's not necessary, but usually this approach is been used in such cases. Please, ignore if not relevant